nixos/modules/services/monitoring/grafana.nix

```diff
@@ -6,6 +6,8 @@ let
   cfg = config.services.grafana;
   opt = options.services.grafana;
   declarativePlugins = pkgs.linkFarm "grafana-plugins" (builtins.map (pkg: { name = pkg.pname; path = pkg; }) cfg.declarativePlugins);
+  useMysql = cfg.database.type == "mysql";
+  usePostgresql = cfg.database.type == "postgres";
```

Comment:
Would be nice if you could check the host this is connecting to as well to really confirm that it is a local connection and this `after` makes sense.

Reply:
oh something like `useLocalPostgresql`, that makes sense!

Reply:
I know there are four (to five) possibilities how to connect to a local postgresql instance:
* /run/postgresql
* /var/run/postgresql
* [::1}:${services.postgresql.port}
* 127.0.0.1:${services.postgresql.port}

and possible the unix socket address could be custom.
I don't really know about mysql.

I don't know if the complexity makes up for the benefits.

Reply:
Valid point. In other services there are options to provision a local database (often called `createDatabaseLocally`) which helps infer this behaviour... but no such option exists here, so maybe you should ignore my point and continue with what you're doing.

Sorry for the hassle :heart: